schema/elasticsearch/visibility/versioned/v1/reindex.sh

```diff
@@ -13,7 +13,27 @@ ES_VIS_INDEX_V1="${ES_VIS_INDEX_V1:-temporal_visibility_v1_dev}"
 CUSTOM_SEARCH_ATTRIBUTES="${CUSTOM_SEARCH_ATTRIBUTES:-[\"CustomKeywordField\",\"CustomStringField\",\"CustomIntField\",\"CustomDatetimeField\",\"CustomDoubleField\",\"CustomBoolField\"]}"
 AUTO_CONFIRM="${AUTO_CONFIRM:-}"
 
+ES_ENDPOINT="${ES_SCHEME}://${ES_SERVER}:${ES_PORT}"
 DIR_NAME="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+echo "=== Step 1. Add custom search attributes to the new index. ==="
+# _doc and ?include_type_name=true for compatibility with ES6.
+CUSTOM_SEARCH_ATTRIBUTES_MAPPING=$(curl --silent --user "${ES_USER}":"${ES_PWD}" "${ES_ENDPOINT}/${ES_VIS_INDEX_V0}/_doc/_mapping?include_type_name=true" | jq -c '.. | .Attr? | select(.!=null) | .properties | with_entries(select(.key == ($customSA[]))) | {properties:.}' --argjson customSA "${CUSTOM_SEARCH_ATTRIBUTES}")
+jq -n "${CUSTOM_SEARCH_ATTRIBUTES_MAPPING}"
+if [ -z "${AUTO_CONFIRM}" ]; then
```

Comment:
nit: prefer `[[` over `[` generally as quoting is easier in many cases

Reply:
Good to know. Leaving it as is for now.